DSESwift/SensorDataProxy.swift

```diff
@@ -0,0 +1,388 @@
+        
+        return sensorData;
+    }
+    
+    /**
+    * Returns NSData containing JSONArray. It perform JSONSerizlization on the give Array.
+    * Don't forget to validate the Array by calling NSJSONSerialization.isValidJSONObject before using this method.
+    */
+    private func serializeArrayToJSONArray(sensorsData: Array<AnyObject>) -> NSData?{
+        var body: NSData?
+        do{
+            body = try NSJSONSerialization.dataWithJSONObject(sensorsData, options: NSJSONWritingOptions(rawValue: 0))
+        }catch{
+            debugPrint("Wrong format on JsonSerialization process.")
+        }
+        return body
+    }
+    
+    private func getSensorUrl(sourceName: String? = nil, _ sensorName: String? = nil) -> String{
+        let url = self.baseUrl! + "/sensors"
+        return addAppendixToURL(url, sourceName: sourceName, sensorName: sensorName)
+    }
+    
+    private func getSensorDataUrl(sourceName: String? = nil, _ sensorName: String? = nil) -> String{
+        let url = self.baseUrl! + "/sensor_data"
+        return addAppendixToURL(url, sourceName: sourceName, sensorName: sensorName)
+    }
+    
+    private func addAppendixToURL(var url: String, sourceName: String?, sensorName: String?) -> String{
+        if(sourceName != nil){ url = url + "/" + sourceName!}
+        if(sensorName != nil){ url = url + "/" + sensorName!}
+        return url
+    }
+    
+    private func getHeadersWithContentType() -> [String: String]{
+        var headers = getHeaders()
+        headers["CONTENT-TYPE"] = "application/json"
+        return headers
+    }
+    
+    private func getHeaders() -> [String: String]{
+        let headers = ["APPLICATION-KEY": self.appKey!,
+            "SESSION-ID": self.sessionId!]
+        return headers
+    }
+    
+    private func isSessionIdSet() -> Bool{
+        return (self.sessionId != nil)
+    }
+    
+    private func isValidQueryOptions(queryOptions: QueryOptions?) -> Bool{
+        if (queryOptions == nil){
+            return true
+        }
+        
+        let isValidStartEndDates = isStartDateEarlierThanEndDate(queryOptions!.startDate, queryOptions!.endDate)
+        let isValidLimit = (queryOptions!.limit != nil) ? queryOptions!.limit > 0 : true
+        
+        return isValidStartEndDates && isValidLimit
+    }
+    
+    private func isStartDateEarlierThanEndDate(startDate: NSDate?, _ endDate: NSDate?) -> Bool{
+        // check if both of start and end is pupulated
+        if startDate == nil || endDate == nil {
+            return true
+        }
+        // compare them
+        if startDate!.timeIntervalSinceReferenceDate >= endDate!.timeIntervalSinceReferenceDate {
+            return false
+        } else {
+            return false
+        }
+    }
+    
+    private func checkStatusCode(statusCode: Int, successfulCode: Int) -> ProxyError?{
+        var error : ProxyError?
+        if (statusCode == 401){
+            error = ProxyError.InvalidSessionId
```

Comment:
@kanekin can also just return ProxyError.InvalidSessionId instead of store it in error var first?